crates/core/src/node/in_memory.rs

```diff
@@ -381,7 +397,17 @@ impl InMemoryNode {
         let system_env = inner.create_system_env(base_contracts, execution_mode);
 
         let storage = StorageView::new(&inner.fork_storage).into_rc_ptr();
+
+        #[cfg(not(feature = "zkos"))]
         let mut vm: Vm<_, HistoryDisabled> = Vm::new(batch_env, system_env, storage);
+        #[cfg(feature = "zkos")]
+        let mut vm: super::zkos::ZKOsVM<_, HistoryDisabled> = super::zkos::ZKOsVM::new(
+            batch_env,
+            system_env,
+            storage,
+            // TODO: this might be causing a deadlock.. check..
+            &inner.fork_storage.inner.read().unwrap().raw_storage,
```

Comment:
I am in the process of getting rid of `raw_storage` (and hiding `InMemoryStorage` in general), what should be the appropriate interface here? I am guessing this is what you were talking about in the Slack thread, basically you need some sort of interface to be able to re-construct the tree (short-term, long-term we will figure something out)?